### PR TITLE
Moduserbl 185 RESET_PASSWORD_* values may conflict with mod-authtoken configuration

### PR DESCRIPTION
**Pull request by SreejaMangarapu**
Purpose: https://folio-org.atlassian.net/browse/MODUSERBL-185
Approach: removed default expiration time and expiration units and made code changes so that the reset link will be expired with the expiration time of the token configured in mod-authtoken

### CODE REVIEW

**Review by Vignesh-kalyanasundaram — APPROVED** [review]

File: src/test/java/org/folio/rest/GeneratePasswordRestLinkTest.java [hunk 1]
```diff
@@ -21,6 +21,8 @@
 import io.vertx.core.json.JsonObject;
 import io.vertx.ext.unit.TestContext;
 import io.vertx.ext.unit.junit.VertxUnitRunner;
+
```
Comment (gurleenkaurbp) [inline src/test/java/org/folio/rest/GeneratePasswordRestLinkTest.java:24]: please remove extra lines

File: src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java [hunk 2]
```diff
@@ -121,91 +112,59 @@ public Future<String> sendPasswordResetLink(String userId, Map<String, String> o
...
-      expirationTime = MAXIMUM_EXPIRATION_TIME;
-      configMapHolder.value.put(LINK_EXPIRATION_TIME_CONFIG_KEY, String.valueOf(MAXIMUM_EXPIRATION_TIME_IN_WEEKS));
-      configMapHolder.value.put(LINK_EXPIRATION_UNIT_OF_TIME_CONFIG_KEY, ExpirationTimeUnit.WEEKS.name().toLowerCase());
-    }
+  public Future<Boolean> isPasswordExists(String userId, String token, Holder<String> tokenHolder, OkapiConnectionParams connectionParams, Holder<String> passwordResetActionIdHolder) {
 
-    passwordResetActionIdHolder.value = UUID.randomUUID().toString();
+    tokenHolder.value = token;
+    JsonObject payload = new JsonObject(Buffer.buffer(Base64.getDecoder().decode(token.split("\\.")[1])));
+    Long exp = payload.getLong("exp");
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java:158]: Why not add the full word instead of exp ?

Comment (gurleenkaurbp) [reply src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java:158]: Lets make it a constant

File: src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java [hunk 3]
```diff
@@ -121,91 +112,59 @@ public Future<String> sendPasswordResetLink(String userId, Map<String, String> o
...
+    Date expirationDate = new Date(exp * 1000);
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java:159]: What is this 1000 value ? Can this be also stated as a constant ?

File: src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java [hunk 4]
```diff
@@ -121,91 +112,59 @@ public Future<String> sendPasswordResetLink(String userId, Map<String, String> o
...
-          .withAdditionalProperty("expirationUnitOfTime", expirationUnitOfTimeFromConfig))
+          .withAdditionalProperty("forgotPasswordLink", forgotPasswordLink))
       .withText(StringUtils.EMPTY)
       .withLang(DEFAULT_NOTIFICATION_LANG);
     return notificationClient.sendNotification(notification, connectionParams);
   }
 
-  private Future<String> signToken(OkapiConnectionParams connectionParams, Boolean passwordExists, Holder<Boolean> passwordExistsHolder, Holder<String> passwordResetActionIdHolder) {
-    passwordExistsHolder.value = passwordExists;
+  private Future<String> signToken(OkapiConnectionParams connectionParams, Holder<String> passwordResetActionIdHolder) {
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java:140]: Please add method java docs, what this method will do etc.

File: src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java [hunk 5]
```diff
@@ -121,91 +112,59 @@ public Future<String> sendPasswordResetLink(String userId, Map<String, String> o
...
-    String expirationUnitOfTimeFromConfig = configMapHolder.value.getOrDefault(
-      LINK_EXPIRATION_UNIT_OF_TIME_CONFIG_KEY, LINK_EXPIRATION_UNIT_OF_TIME_DEFAULT);
-
-    long expirationTime = convertDateToMilliseconds(expirationTimeFromConfig, expirationUnitOfTimeFromConfig);
-    if (expirationTime > MAXIMUM_EXPIRATION_TIME) {
-      expirationTime = MAXIMUM_EXPIRATION_TIME;
-      configMapHolder.value.put(LINK_EXPIRATION_TIME_CONFIG_KEY, String.valueOf(MAXIMUM_EXPIRATION_TIME_IN_WEEKS));
-      configMapHolder.value.put(LINK_EXPIRATION_UNIT_OF_TIME_CONFIG_KEY, ExpirationTimeUnit.WEEKS.name().toLowerCase());
-    }
+  public Future<Boolean> isPasswordExists(String userId, String token, Holder<String> tokenHolder, OkapiConnectionParams connectionParams, Holder<String> passwordResetActionIdHolder) {
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java:154]: Method Java docs please.

**Review by gurleenkaurbp — COMMENTED** [review]
Please add necessary logs as per the standards. Its unfortunate that this module has not been visited with best logging practice but we can address some of it as part of this PR.

File: src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java (hunk 2 above)
Comment (SreejaMangarapu) [reply src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java:158]: In the token payload we get expiration time as **exp**, so I have used exp for indicating expiration of token

File: src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java (hunk 3 above)
Comment (SreejaMangarapu) [reply src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java:159]: here we are converting seconds to milliseconds which is required for date object, probably constant is not required for this

**Review by gurleenkaurbp — APPROVED** [review]

File: src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java [hunk 6]
```diff
@@ -121,34 +117,35 @@ public Future<String> sendPasswordResetLink(String userId, Map<String, String> o
...
-    String expirationTimeFromConfig =
-      configMapHolder.value.getOrDefault(LINK_EXPIRATION_TIME_CONFIG_KEY, LINK_EXPIRATION_TIME_DEFAULT);
-    String expirationUnitOfTimeFromConfig = configMapHolder.value.getOrDefault(
-      LINK_EXPIRATION_UNIT_OF_TIME_CONFIG_KEY, LINK_EXPIRATION_UNIT_OF_TIME_DEFAULT);
+    String forgotPasswordLinkHost = configMapHolder.value.getOrDefault(FOLIO_HOST_CONFIG_KEY, FOLIO_HOST_DEFAULT);
+    String forgotPasswordLinkPath = configMapHolder.value.getOrDefault(FORGOT_PASSWORD_UI_PATH_CONFIG_KEY, forgotPasswordUIPathDefault);
+    String forgotPasswordLink = forgotPasswordLinkHost + forgotPasswordLinkPath;
 
-    String eventConfigName = passwordExistsHolder.value ? RESET_PASSWORD_EVENT_CONFIG_NAME : CREATE_PASSWORD_EVENT_CONFIG_NAME;
+    boolean passwordExistsValue = (null != passwordExists) && passwordExists;
```
[the remark this answers is not in the document]
Comment (SreejaMangarapu) [reply src/main/java/org/folio/service/PasswordResetLinkServiceImpl.java]: replaced with **boolean passwordExistsValue = Boolean.TRUE.equals(passwordExists);**

**Pull request merged**